SQL: drop BASE TABLE type in favour for just TABLE

bpintea opened the pull request:
This PR drops the table type `BASE TABLE` for indices and replaces
all occurrences with just `TABLE`, since his type is widely-used and
friendlier to the client applications that query for certain table types
in their discovery mode.

The `TABLE` type is also explicitly mentioned by the JDBC and ODBC
standards and although other data source-specific types are permitted,
older apps will not work well with them.

bpintea committed:
Drop BASE TABLE type in favour for just TABLE

This commit drops the table type 'BASE TABLE' and replaces all
occurences with just 'TABLE', since his type is wider-used and
friendlier to the client applications that query for certain table types
in their discovery mode.

The 'TABLE' type is also explicitely mentioned by the JDBC and ODBC
standards and although other data source-specific types are permitted,
older apps will not work well with them.

bpintea committed:
Update tests, rename BASE TABLE to TABLE

Update tests, rename 'BASE TABLE' to 'TABLE'.

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/drop_base_table

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/drop_base_table

bpintea commented:
@elasticmachine run elasticsearch-ci/bwc

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into enh/drop_base_table

elasticmachine commented:
Pinging @elastic/es-ql (:Query Languages/SQL)

matriv reviewed (approved):
LGTM

costin commented on x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java:
```diff
@@ -76,13 +76,12 @@
         UNKNOWN("UNKNOWN", "UNKNOWN");
 
-        public static final String SQL_BASE_TABLE = "BASE TABLE";
         public static final String SQL_TABLE = "TABLE";
```
Since these constants are used somewhere else, it makes sense to move them outside `IndexType` and reuse them inside the enum declaration as well.

costin commented on x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysTablesTests.java:
```diff
@@ -105,9 +107,9 @@ public void testSysTablesCatalogNoTypes() throws Exception {
         executeCommand("SYS TABLES CATALOG LIKE '' LIKE '' ", r -> {
             assertEquals(2, r.size());
-            assertEquals("BASE TABLE", r.column(3));
+            assertEquals(SQL_TABLE, r.column(3));
```
👍

costin reviewed (approved):
LGTM

astefan reviewed (approved):
LGTM

bpintea committed:
Refactor table type constants out of IndexType

Move SQL_TABLE/_ALIAS out of IndexType, so that they can also be used in
that Enum definition.

bpintea commented:
@elasticmachine run elasticsearch-ci/bwc elasticsearch-ci/2

bpintea commented:
@elasticmachine run elasticsearch-ci/2
